### PR TITLE
cargo_toml: Do not overwrite system-deps version if already set

### PR DESCRIPTION
**Pull request by MarijnS95**
For [1] we need to override the target library version because 1.20 does not exist yet. Gir keeps overwriting this value making it impossible to check such changes in.

As cleanup remove the unnecessary get_mut in set_string: insert already replaces an existing value if the given key exists.

[1]: https://gitlab.freedesktop.org/gstreamer/gstreamer-rs/-/merge_requests/649#note_723814

### CODE REVIEW

File: src/codegen/sys/cargo_toml.rs
Comment (sdroege) [inline src/codegen/sys/cargo_toml.rs:121]: Doesn't this basically revert https://github.com/gtk-rs/gir/pull/1022 ?

Comment (MarijnS95) [reply src/codegen/sys/cargo_toml.rs:121]: This time I fixed my mistake from #1011. Since that PR didn't bother to state an actual bug has been fixed _nor how_, I couldn't be bothered to explain either.

Hint: the difference is already in the signature for `fn entry` vs `fn get_mut`.

Comment (GuillaumeGomez) [reply src/codegen/sys/cargo_toml.rs:121]: Please add a code comment explaining all this so that we know exactly what is going on in case we need to update it once again in the future.

Comment (MarijnS95) [reply src/codegen/sys/cargo_toml.rs:122]: @GuillaumeGomez Added an extra comment explaining this allows overriding `name` and `version` by hand. More explanation and links are in the commit, unless you want those inline as well.

I doubt you intend to explain the difference between `.get_mut` and `.entry` in an inline comment though, that is better suited for the commit addressing the bug itself.

Comment (sdroege) [reply src/codegen/sys/cargo_toml.rs:122]: > Since that PR didn't bother to state an actual bug has been fixed nor how, I couldn't be bothered to explain either.

You mean #1022? The full explanation is in the linked issue #1020 :)

Comment (MarijnS95) [reply src/codegen/sys/cargo_toml.rs:122]: Not that I can see, and even if it did, such an explanation does not show up in `git blame` nor `git show`.

Comment (sdroege) [reply src/codegen/sys/cargo_toml.rs:122]: As #1020 explains, autogenerating the -sys crate put a dependency with a wrong pkg-config name into `Cargo.toml`: `tracker-sparql-3.0` it should've been, `tracker_sparql_3.0` it was. That's what #1022 fixed.

Comment (MarijnS95) [reply src/codegen/sys/cargo_toml.rs:122]: @sdroege Thanks for providing a perfect example to the problem (lacking commit titles/messages) I'm highlighting here.

My mistake in #1011 causes no `name = "name-with-hyphens"` override to show up _at all_, because `get_mut` does not allow modifying the underlying map with insertions (it returns `Option<&mut ItemType>`, so only a mutable reference to the item _if it exists_). That's what `fn entry` is for.
Note that `set_string` - reintroduced by #1022 - does not perform this conversion - `lib_name` is already "canonicalized" with hyphens.

I think the author is seeing underscores because it falls back to `[lib] name = "name_with_underscores"`, again because the unversioned `[...system-deps...]` table is generated without `name` entry.

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by deanleggo** [comment]
Hey @MarijnS95, I'm new to this and I didn't know my #1022 completely overrides your #1011 (from a week or two before). Sorry, next time I'll check the history and make sure the PR and commit message states why the change was made (not relying on a linked issue/discussion) and discuss with the author of the code about the change.

**Comment by MarijnS95** [comment]
@deanleggo No worries, you didn't override #1022 but rather fix a serious regression introduced by it. However, according to the  discussion in #1020 and subsequent changes made in #1022 it seems that the underlying issue was not understood. This is why I advocate for well thought out commit titles (and descriptions where necessary).

Furthermore using git blame to find an eventual isolated change to those lines might help understand the problem and find the previous author, though in this case the maintainers likely remembered who (what PR) changed it just under 2 weeks ago. CC-ing that author is never a bad idea so that they are aware of the change, are able to explain their situation/use-case and most importantly: learn from their mistakes.

In this PR I have applied an alternate solution, the solution I intended to write in #1011 and would have suggested if I were aware of #1022 before it got merged. No harm done, just that these small and nasty regressions are a great learning source for everyone.

**Pull request merged**
